### ISSUE TITLE
Build process

### ISSUE THREAD

**Issue by xyeres** [digitoimistodude/air-light#164]
Hi team dude! I really like the theme, so thank you for the work you're constantly putting into it. 

I wanted to check in and ask your input on the build process for production. Trying to keep this short:

I'm coming from a background using javascript stacks, (react, nextJs, etc). With these other stacks, the bundler typically creates a `build` folder as a the final artifact that is then deployed on the server. This creates a nice separation between development and production files.   

I am new to WP theming and modern WP workflows, so bear with me, but I am a little confused on how I could approach the build process with air-light. It seems that running `gulp` builds the files for both dev and prod in their respective directories, all co-mingled. And then the WP config sets the environment to `production` and that is what properly enqueues the js, css, etc. 

When I go to build my CI/CD pipeline, I prefer not to commit or track any build files to git, but with them all co-mingled this way it seems a little messy to `gitignore` them file by file. 

**In regard to how air-light is setup, is it commonplace to commit everything in the theme folder to git (build files and all) and then deploy this to a WP instance?** 

Or does it make more sense to ignore all the build files, only commit source and then in the build pipeline, run gulp and only use the prod files? (this seems in conflict with how air-light is setup but maybe I don't understand)

I hope my question doesn't sound super general, I'm hoping to understand how air-light was designed to be used

Thanks so much!

**Comment by ronilaukkarinen** [comment]
@xyeres Thanks for the interest towards our build process and workflow. I try to explain this best I can.

Air-light is only a part of the process and this repository is of course built from our approach first, then the other use cases. 

In WordPress development here at Dude

1. We don't use theme as its own repository in the WordPress projects. Instead we use [dudestack](https://github.com/digitoimistodude/dudestack) as stack where themes are only part of the project
2. We don't use push-to-deploy. Instead we use [capistrano](https://github.com/capistrano/capistrano) which gets the whole project from git instead of just a theme. For this we need some development files (composer etc.) because we may want to run some tests in staging as well. This is also why production files are not ignored as they are pulled from git.
3. Our project development files are also in their own separate repository [devpackages](https://github.com/digitoimistodude/devpackages) and are used in the root level of the project, <b>not</b> in the theme dir
4. We use starter scripts that bundle all these together
5. We only use air-light as its own repo when we continue development of air-light itself, as described in [Air development](https://github.com/digitoimistodude/air-light#air-development)

This being said, it's rather complicated to get the "full Dude setup" run as intented on other machines, this is well documented for us internally but for public we just offer all the repos and DIY Wiki [here](https://github.com/digitoimistodude/dudestack/wiki) (something may be always outdated, contributions welcome).

As for "build" dirs and such, we really appreciate the original way of developing WordPress without any "fancy" dirs or stuff like "views", "assets" etc., or fancy templating languages. You can read some of our rationale for this [here](https://github.com/digitoimistodude/dudestack#how-its-different-why-should-i-use-this). We originally based our stack to [roots/bedrock](https://github.com/roots/bedrock) +  [roots/sage](https://github.com/roots/sage) but differed in a point where they took it more of an "app-like" direction. So if you'd like more templating, more build-centered and more app-like experience, you should try bedrock's stuff. Our stack is a combination of traditional WordPress development and morern tools.

> In regard to how air-light is setup, is it commonplace to commit everything in the theme folder to git (build files and all) and then deploy this to a WP instance?

So, in summary: Create a WordPress setup (with dudestack or your preferred way) then commit the whole project _with the theme inside_. This is at least what we prefer. I understand it may not work with Local by Flywheel or other build systems but this is how we do it. Do what works best for you and have fun! 😉

**Comment by xyeres** [comment]
Hey thanks @ronilaukkarinen for taking the time to explain this, really insightful, I tried my best to understand how Dude uses air-light and this question kept coming up for me so I hope others find this helpful too.

**Comment by ronilaukkarinen** [comment]
You're welcome! I understand it's not the most clear for many. This theme is highly opinionated but at the same time attempts to be ultra minimal. Most of the themes are filled with all kinds of bloat that are often not needed. Our approach is the contrary. In my mind devs should not start "cleaning" first, they should be right at it, creating speedy websites, that is. For this reason we offer the repositories as-is. Our way is not the only answer.

Please also see: **git pull new changes to my custom theme #87**

**Comment by xyeres** [comment]
That simplicity and low code overhead is what drew us to use air-light in the beginning!
